File: LeetCode/Easy/0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n=nums.size();
        unordered_map<int,int> mp;

        for(int i=0;i<n;i++){
            
            int val=nums[i];
            int compliment=target-val;

            if(mp.find(compliment) != mp.end() && mp[compliment]!= i)
                return {i,mp[compliment]};

            mp[val]=i;
        }
        return {};
    }
};

File: Python3/Easy/0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mp={}
        for i, val in enumerate(nums):
            comp=target - val
            if comp in mp:
                return [mp[comp],i]

            mp[val]=i
        
        return []
